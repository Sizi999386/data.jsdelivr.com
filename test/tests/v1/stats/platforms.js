const { expect } = require('chai');
const { makeEndpointSnapshotTests, makeEndpointPaginationTests, makePaginatedEndpointAssertions, setupSnapshots } = require('../../../utils');

describe('/v1/stats/platforms', () => {
	before(() => {
		setupSnapshots(__filename);
	});

	makeStatsPlatformsTests();
});

function makeStatsPlatformsTests () {
	makePaginatedEndpointAssertions('/v1/stats/platforms{?continent,country,period}', {}, [
		{
			params: { period: '2020-04' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.sumBy(response.body, 'prev.share')).to.be.closeTo(91.74, .1);
				expect(response.body).to.have.lengthOf(14);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2020-04', continent: 'EU' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.sumBy(response.body, 'prev.share')).to.be.closeTo(91.74, .1);
				expect(response.body).to.have.lengthOf(14);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2020-04', country: 'PL' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.sumBy(response.body, 'prev.share')).to.be.closeTo(91.74, .1);
				expect(response.body).to.have.lengthOf(14);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2020-02' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(20);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2020-02', continent: 'EU' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(20);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2020-02', country: 'PL' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(20);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: 's-month' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(8);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: 's-month', continent: 'EU' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(8);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: 's-month', country: 'AE' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(8);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2020' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(20);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2020', continent: 'EU' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(20);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2020', country: 'PL' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.every(response.body, result => result.prev.share === null)).to.be.true;
				expect(response.body).to.have.lengthOf(20);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: 's-year' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.sumBy(response.body, 'prev.share')).to.be.closeTo(4.54, .1);
				expect(response.body).to.have.lengthOf(8);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: 's-year', continent: 'EU' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.sumBy(response.body, 'prev.share')).to.be.closeTo(4.54, .1);
				expect(response.body).to.have.lengthOf(8);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: 's-year', country: 'AE' },
			assert: (response) => {
				expect(_.sumBy(response.body, 'share')).to.be.closeTo(100, .1);
				expect(_.sumBy(response.body, 'prev.share')).to.be.closeTo(4.54, .1);
				expect(response.body).to.have.lengthOf(8);
				expect(response).to.matchSnapshot();
			},
		},
		{
			params: { period: '2019' },
			assert: (response) => {
				expect(response.body).to.be.empty;
			},
		},
	]);

	makeEndpointPaginationTests('/v1/stats/platforms', { period: '2020-04' });

	makeEndpointSnapshotTests('/v1/stats/platforms{?continent,country,period}', {}, [
		{ continent: 'X' },
		{ country: 'X' },
		{ continent: 'EU', country: 'PL' },
		{ period: 'month' },
	], { status: 400 });
}
