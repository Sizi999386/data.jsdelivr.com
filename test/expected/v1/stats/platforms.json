{
	"/v1/stats/platforms?continent=EU&country=PL": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Invalid parameter value: `period` must be one of [s-month, s-year] or a valid date in one of the following ISO formats [YYYY, YYYY-MM], object contains a conflict between optional exclusive peers [continent, country]."
	},
	"/v1/stats/platforms?continent=EU&period=2020": [
		{
			"name": "platform 19",
			"share": 13.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 9",
			"share": 13.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 18",
			"share": 11.01,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 8",
			"share": 11.01,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 17",
			"share": 8.61,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 7",
			"share": 8.61,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 16",
			"share": 6.51,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 6",
			"share": 6.51,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 15",
			"share": 4.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 5",
			"share": 4.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 14",
			"share": 3.19,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 4",
			"share": 3.19,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 1.42,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 1.42,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 12",
			"share": 0.58,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 0.58,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 0.23,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 0.23,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 0.04,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 0.04,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?continent=EU&period=2020-02": [
		{
			"name": "platform 19",
			"share": 13.56,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 9",
			"share": 13.56,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 18",
			"share": 10.86,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 8",
			"share": 10.86,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 17",
			"share": 8.45,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 7",
			"share": 8.45,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 16",
			"share": 6.35,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 6",
			"share": 6.35,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 15",
			"share": 4.55,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 5",
			"share": 4.55,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 14",
			"share": 3.05,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 4",
			"share": 3.05,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 1.85,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 1.85,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 12",
			"share": 0.95,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 0.95,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 0.34,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 0.34,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 0.04,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 0.04,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?continent=EU&period=2020-04": [
		{
			"name": "platform 19",
			"share": 14.26,
			"prev": {
				"share": 9.66
			}
		},
		{
			"name": "platform 9",
			"share": 14.26,
			"prev": {
				"share": 9.66
			}
		},
		{
			"name": "platform 18",
			"share": 11.42,
			"prev": {
				"share": 8.62
			}
		},
		{
			"name": "platform 8",
			"share": 11.42,
			"prev": {
				"share": 8.62
			}
		},
		{
			"name": "platform 17",
			"share": 8.89,
			"prev": {
				"share": 7.59
			}
		},
		{
			"name": "platform 7",
			"share": 8.89,
			"prev": {
				"share": 7.59
			}
		},
		{
			"name": "platform 16",
			"share": 6.68,
			"prev": {
				"share": 6.55
			}
		},
		{
			"name": "platform 6",
			"share": 6.68,
			"prev": {
				"share": 6.55
			}
		},
		{
			"name": "platform 15",
			"share": 4.78,
			"prev": {
				"share": 5.52
			}
		},
		{
			"name": "platform 5",
			"share": 4.78,
			"prev": {
				"share": 5.52
			}
		},
		{
			"name": "platform 14",
			"share": 3.2,
			"prev": {
				"share": 4.48
			}
		},
		{
			"name": "platform 4",
			"share": 3.2,
			"prev": {
				"share": 4.48
			}
		},
		{
			"name": "platform 13",
			"share": 0.77,
			"prev": {
				"share": 3.45
			}
		},
		{
			"name": "platform 3",
			"share": 0.77,
			"prev": {
				"share": 3.45
			}
		}
	],
	"/v1/stats/platforms?continent=EU&period=s-month": [
		{
			"name": "platform 12",
			"share": 25.76,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 25.76,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 13.64,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 13.64,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 9.39,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 9.39,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 1.21,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 1.21,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?continent=EU&period=s-year": [
		{
			"name": "platform 12",
			"share": 23.33,
			"prev": {
				"share": 0.58
			}
		},
		{
			"name": "platform 2",
			"share": 23.33,
			"prev": {
				"share": 0.58
			}
		},
		{
			"name": "platform 1",
			"share": 13.33,
			"prev": {
				"share": 0.23
			}
		},
		{
			"name": "platform 11",
			"share": 13.33,
			"prev": {
				"share": 0.23
			}
		},
		{
			"name": "platform 13",
			"share": 10,
			"prev": {
				"share": 1.42
			}
		},
		{
			"name": "platform 3",
			"share": 10,
			"prev": {
				"share": 1.42
			}
		},
		{
			"name": "platform 0",
			"share": 3.33,
			"prev": {
				"share": 0.04
			}
		},
		{
			"name": "platform 10",
			"share": 3.33,
			"prev": {
				"share": 0.04
			}
		}
	],
	"/v1/stats/platforms?continent=X": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Invalid parameter value: `period` must be one of [s-month, s-year] or a valid date in one of the following ISO formats [YYYY, YYYY-MM], `continent` must be a valid continent code in uppercase."
	},
	"/v1/stats/platforms?country=AE&period=s-month": [
		{
			"name": "platform 12",
			"share": 25.76,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 25.76,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 13.64,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 13.64,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 9.39,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 9.39,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 1.21,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 1.21,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?country=AE&period=s-year": [
		{
			"name": "platform 12",
			"share": 23.33,
			"prev": {
				"share": 0.58
			}
		},
		{
			"name": "platform 2",
			"share": 23.33,
			"prev": {
				"share": 0.58
			}
		},
		{
			"name": "platform 1",
			"share": 13.33,
			"prev": {
				"share": 0.23
			}
		},
		{
			"name": "platform 11",
			"share": 13.33,
			"prev": {
				"share": 0.23
			}
		},
		{
			"name": "platform 13",
			"share": 10,
			"prev": {
				"share": 1.42
			}
		},
		{
			"name": "platform 3",
			"share": 10,
			"prev": {
				"share": 1.42
			}
		},
		{
			"name": "platform 0",
			"share": 3.33,
			"prev": {
				"share": 0.04
			}
		},
		{
			"name": "platform 10",
			"share": 3.33,
			"prev": {
				"share": 0.04
			}
		}
	],
	"/v1/stats/platforms?country=PL&period=2020": [
		{
			"name": "platform 19",
			"share": 13.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 9",
			"share": 13.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 18",
			"share": 11.01,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 8",
			"share": 11.01,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 17",
			"share": 8.61,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 7",
			"share": 8.61,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 16",
			"share": 6.51,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 6",
			"share": 6.51,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 15",
			"share": 4.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 5",
			"share": 4.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 14",
			"share": 3.19,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 4",
			"share": 3.19,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 1.42,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 1.42,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 12",
			"share": 0.58,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 0.58,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 0.23,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 0.23,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 0.04,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 0.04,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?country=PL&period=2020-02": [
		{
			"name": "platform 19",
			"share": 13.56,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 9",
			"share": 13.56,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 18",
			"share": 10.86,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 8",
			"share": 10.86,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 17",
			"share": 8.45,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 7",
			"share": 8.45,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 16",
			"share": 6.35,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 6",
			"share": 6.35,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 15",
			"share": 4.55,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 5",
			"share": 4.55,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 14",
			"share": 3.05,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 4",
			"share": 3.05,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 1.85,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 1.85,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 12",
			"share": 0.95,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 0.95,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 0.34,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 0.34,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 0.04,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 0.04,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?country=PL&period=2020-04": [
		{
			"name": "platform 19",
			"share": 14.26,
			"prev": {
				"share": 9.66
			}
		},
		{
			"name": "platform 9",
			"share": 14.26,
			"prev": {
				"share": 9.66
			}
		},
		{
			"name": "platform 18",
			"share": 11.42,
			"prev": {
				"share": 8.62
			}
		},
		{
			"name": "platform 8",
			"share": 11.42,
			"prev": {
				"share": 8.62
			}
		},
		{
			"name": "platform 17",
			"share": 8.89,
			"prev": {
				"share": 7.59
			}
		},
		{
			"name": "platform 7",
			"share": 8.89,
			"prev": {
				"share": 7.59
			}
		},
		{
			"name": "platform 16",
			"share": 6.68,
			"prev": {
				"share": 6.55
			}
		},
		{
			"name": "platform 6",
			"share": 6.68,
			"prev": {
				"share": 6.55
			}
		},
		{
			"name": "platform 15",
			"share": 4.78,
			"prev": {
				"share": 5.52
			}
		},
		{
			"name": "platform 5",
			"share": 4.78,
			"prev": {
				"share": 5.52
			}
		},
		{
			"name": "platform 14",
			"share": 3.2,
			"prev": {
				"share": 4.48
			}
		},
		{
			"name": "platform 4",
			"share": 3.2,
			"prev": {
				"share": 4.48
			}
		},
		{
			"name": "platform 13",
			"share": 0.77,
			"prev": {
				"share": 3.45
			}
		},
		{
			"name": "platform 3",
			"share": 0.77,
			"prev": {
				"share": 3.45
			}
		}
	],
	"/v1/stats/platforms?country=X": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Invalid parameter value: `period` must be one of [s-month, s-year] or a valid date in one of the following ISO formats [YYYY, YYYY-MM], `country` must be a valid ISO 3166-1 alpha-2 country code in uppercase."
	},
	"/v1/stats/platforms?period=2020": [
		{
			"name": "platform 19",
			"share": 13.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 9",
			"share": 13.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 18",
			"share": 11.01,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 8",
			"share": 11.01,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 17",
			"share": 8.61,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 7",
			"share": 8.61,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 16",
			"share": 6.51,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 6",
			"share": 6.51,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 15",
			"share": 4.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 5",
			"share": 4.7,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 14",
			"share": 3.19,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 4",
			"share": 3.19,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 1.42,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 1.42,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 12",
			"share": 0.58,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 0.58,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 0.23,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 0.23,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 0.04,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 0.04,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?period=2020-02": [
		{
			"name": "platform 19",
			"share": 13.56,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 9",
			"share": 13.56,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 18",
			"share": 10.86,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 8",
			"share": 10.86,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 17",
			"share": 8.45,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 7",
			"share": 8.45,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 16",
			"share": 6.35,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 6",
			"share": 6.35,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 15",
			"share": 4.55,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 5",
			"share": 4.55,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 14",
			"share": 3.05,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 4",
			"share": 3.05,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 1.85,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 1.85,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 12",
			"share": 0.95,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 0.95,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 0.34,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 0.34,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 0.04,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 0.04,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?period=2020-04": [
		{
			"name": "platform 19",
			"share": 14.26,
			"prev": {
				"share": 9.66
			}
		},
		{
			"name": "platform 9",
			"share": 14.26,
			"prev": {
				"share": 9.66
			}
		},
		{
			"name": "platform 18",
			"share": 11.42,
			"prev": {
				"share": 8.62
			}
		},
		{
			"name": "platform 8",
			"share": 11.42,
			"prev": {
				"share": 8.62
			}
		},
		{
			"name": "platform 17",
			"share": 8.89,
			"prev": {
				"share": 7.59
			}
		},
		{
			"name": "platform 7",
			"share": 8.89,
			"prev": {
				"share": 7.59
			}
		},
		{
			"name": "platform 16",
			"share": 6.68,
			"prev": {
				"share": 6.55
			}
		},
		{
			"name": "platform 6",
			"share": 6.68,
			"prev": {
				"share": 6.55
			}
		},
		{
			"name": "platform 15",
			"share": 4.78,
			"prev": {
				"share": 5.52
			}
		},
		{
			"name": "platform 5",
			"share": 4.78,
			"prev": {
				"share": 5.52
			}
		},
		{
			"name": "platform 14",
			"share": 3.2,
			"prev": {
				"share": 4.48
			}
		},
		{
			"name": "platform 4",
			"share": 3.2,
			"prev": {
				"share": 4.48
			}
		},
		{
			"name": "platform 13",
			"share": 0.77,
			"prev": {
				"share": 3.45
			}
		},
		{
			"name": "platform 3",
			"share": 0.77,
			"prev": {
				"share": 3.45
			}
		}
	],
	"/v1/stats/platforms?period=month": {
		"date": "2010-12-31",
		"status": 400,
		"message": "Invalid parameter value: `period` must be one of [s-month, s-year] or a valid date in one of the following ISO formats [YYYY, YYYY-MM]."
	},
	"/v1/stats/platforms?period=s-month": [
		{
			"name": "platform 12",
			"share": 25.76,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 2",
			"share": 25.76,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 13",
			"share": 13.64,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 3",
			"share": 13.64,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 1",
			"share": 9.39,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 11",
			"share": 9.39,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 0",
			"share": 1.21,
			"prev": {
				"share": null
			}
		},
		{
			"name": "platform 10",
			"share": 1.21,
			"prev": {
				"share": null
			}
		}
	],
	"/v1/stats/platforms?period=s-year": [
		{
			"name": "platform 12",
			"share": 23.33,
			"prev": {
				"share": 0.58
			}
		},
		{
			"name": "platform 2",
			"share": 23.33,
			"prev": {
				"share": 0.58
			}
		},
		{
			"name": "platform 1",
			"share": 13.33,
			"prev": {
				"share": 0.23
			}
		},
		{
			"name": "platform 11",
			"share": 13.33,
			"prev": {
				"share": 0.23
			}
		},
		{
			"name": "platform 13",
			"share": 10,
			"prev": {
				"share": 1.42
			}
		},
		{
			"name": "platform 3",
			"share": 10,
			"prev": {
				"share": 1.42
			}
		},
		{
			"name": "platform 0",
			"share": 3.33,
			"prev": {
				"share": 0.04
			}
		},
		{
			"name": "platform 10",
			"share": 3.33,
			"prev": {
				"share": 0.04
			}
		}
	]
}
